lib/hd/private.js

```diff
@@ -79,6 +81,13 @@ class HDPrivateKey {
     assert(Buffer.isBuffer(options.chainCode));
     assert(Buffer.isBuffer(options.privateKey));
 
+    if (options.purpose) {
+      assert(common.purposes[options.purpose] !== undefined, 'Bad purpose');
+      this.purpose = options.purpose;
+    } else {
+      this.purpose = 'x';
```

Comment:
Isn't it redundant to set `this.purpose = 'x'` here when its already initialized in the constructor?

Reply:
👍